portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java

```diff
@@ -23,16 +28,29 @@
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         String page = request.getParameter("page");
+        String text = request.getParameter("text");
+        String author = request.getParameter("author");
+        String emotion = request.getParameter("emotion");
+
```

Comment:
I don't intend that to happen but current solution allows it. The reason I skipped the date parameter was because the format of the date was changed, and would be hard to compare. I was just relying on the fact that two comments made on different dates would not be exactly same (which is not very good assumption). I am thinking of refactoring the code and assigning unique id to each comment, so I can use that instead of finding comment based on many parameters.

Reply:
I will refactor this with authentication because that will also require to change things.

Reply:
The unique id is a good idea and requires less storage than all the comment attributes. To generate unique IDs, I would use UUID objects (https://docs.oracle.com/javase/8/docs/api/java/util/UUID.html).

For dates: you could first parse the string into a Date object and then use the .compareTo function to compare Dates with each other. Looks like the DateFormat class has a parse function: https://docs.oracle.com/javase/7/docs/api/java/text/DateFormat.html#parse(java.lang.String)

Reply:
Thanks, I will check out UUID objects. 
I changed date text so that it only shows day, month and year (script.js line 33)  . So if in the DB I have 2020-06-05 (18:50:14.591) EDT, date string would be just 06/05/2020 which I think is harder to parse into Date object

Reply:
+1 to adding unique IDs